### PR TITLE
fix(suite): protocol

### LINKED ISSUE #14813: Missing prefixes of new coins in protocol.ts
**Describe the bug**
Missing prefixes of new coins in `protocol.ts`
If you try to scan QR code that include coin name in receive prefix it will fail.
`QR code contains invalid address for this account`

**Steps to reproduce:**
1. a
2. b
3. c

**Info:**
 - Suite version: desktop 24.10.1 (3cbe786b5c8effd03b331faec20a41ef01d6671b)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.10.1 Chrome/126.0.6478.234 Electron/31.6.0 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.4 regular (revision d71d9e9c343ad04a7f28d6b02f8d3f8945af30d6)
 - Transport: BridgeTransport 3.0.0-bundled.24.10.1

**Screenshots:**
image

### PR DESCRIPTION
**Pull request by enjojoy**
## Description

There was an issue that not all networks had a protocol, now typescript will check for each network to have a protocol prefix assigned, and if a new Network will be added, it will throw a type error that it doesn't have a protocol prefix assigned. 

Also now it's possible to have a list of possible prefixes for each NetworkSymbol.

As a result it's possible to add an address from a QR code for each network that is enabled in suite. 

Added error handling for unknown prefixes and logging to sentry.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/14813

## Screenshots

Error on unknown prefix

image

### CODE REVIEW

File: suite-common/suite-utils/src/protocol.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Protocol, NETWORK_TO_PROTOCOLS } from '@suite-common/suite-constants';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+
+export type ProtocolToNetwork = {
+    [P in Protocol]: NetworkSymbol;
+};
+
+export const PROTOCOL_TO_NETWORK = {} as ProtocolToNetwork;
```
Comment (tomasklim) [inline suite-common/suite-utils/src/protocol.ts]:
```ts
export const getNetworkSymbolForProtocol = (protocol: Protocol): NetworkSymbol | undefined => {
    for (const networkSymbol in NETWORK_TO_PROTOCOLS) {
        const protocols = NETWORK_TO_PROTOCOLS[networkSymbol as NetworkSymbol];
        if (protocols.includes(protocol)) {
            return networkSymbol as NetworkSymbol;
        }
    }

    return undefined;
};
```

File: suite-common/suite-constants/src/protocol.ts [hunk 2]
```diff
@@ -1,36 +1,57 @@
...
+    sol: ['solana'],
+    pol: ['polygon', 'matic'],
+    bnb: ['binance'],
+    op: ['optimism'],
+    test: ['test'],
+    regtest: ['regtest'],
+    tsep: ['sepolia'],
+    thol: ['holesky'],
+    txrp: ['testxrp'],
+    tada: ['testcardano'],
```
Comment (tomasklim) [inline suite-common/suite-constants/src/protocol.ts]: same as network symbol

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: suite-common/suite-constants/src/protocol.ts [hunk 3]
```diff
@@ -1,36 +1,80 @@
...
+    pol: ['polygon', 'matic'],
+    bnb: ['binance', 'bnb'],
+    op: ['optimism', 'op'],
+    test: ['test'],
+    regtest: ['regtest'],
+    tsep: ['sepolia', 'tsep'],
+    thol: ['holesky', 'thol'],
+    txrp: ['testxrp', 'txrp'],
+    tada: ['testcardano', 'tada'],
+    dsol: ['solana', 'sol', 'dsol'],
```
Comment (tomasklim) [inline suite-common/suite-constants/src/protocol.ts]:
```suggestion
    dsol: ['dsol'],
```

File: suite-common/suite-constants/src/protocol.ts [hunk 4]
```diff
@@ -1,36 +1,80 @@
...
+    doge: ['dogecoin', 'doge'],
+    nmc: ['namecoin', 'nmc'],
+    vtc: ['vertcoin', 'vtc'],
+    zec: ['zcash', 'zec'],
+    eth: ['ethereum', 'eth'],
+    etc: ['ethclassic', 'etc'],
+    xrp: ['ripple', 'xrp'],
+    ada: ['cardano', 'ada'],
+    sol: ['solana', 'sol'],
+    pol: ['polygon', 'matic'],
```
Comment (tomasklim) [inline suite-common/suite-constants/src/protocol.ts]:
```suggestion
    pol: ['polygon', 'matic', 'pol'],
```

File: suite-common/suite-constants/src/protocol.ts [hunk 5]
```diff
@@ -1,36 +1,80 @@
...
+    bnb: ['binance', 'bnb'],
```
Comment (tomasklim) [inline suite-common/suite-constants/src/protocol.ts]:
```suggestion
    bnb: ['binance', 'bnb', 'bsc'],
```

File: suite-common/suite-constants/src/protocol.ts [hunk 6]
```diff
@@ -1,36 +1,80 @@
...
+    op: ['optimism', 'op'],
+    test: ['test'],
+    regtest: ['regtest'],
+    tsep: ['sepolia', 'tsep'],
```
Comment (tomasklim) [inline suite-common/suite-constants/src/protocol.ts]: please only symbol for testnets, no sepolia, no holesky,...

**Review by tomasklim — CHANGES_REQUESTED** [review]

**Comment by enjojoy** [comment]
@trezor/qa To test it you can do few following things: 

### 1
When in web app, go to send section and scan QR code with a prefix of picked network. You can generate QR code in some mobile app and enter the address with network prefix as text. 

For example "binance:0x12312312313" (but the address has to be correct)

If the prefix is unknown, there will be an error toast
If the prefix belongs to another known network, there will be an error toast with the network it belongs to 
If the prefix is known and the network is right it should fill in the address

### 2 

Open app through URI protocol scheme  bitcoin:12313

**Review by tomasklim — APPROVED** [review]

**Pull request merged**